File: sesa-backend/src/main/java/com/sesa/salud/dto/auth/LoginResponse.java
/**
 * Autor: Ing. J Sebastian Vargas S
 */

package com.sesa.salud.dto.auth;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LoginResponse {

    private String accessToken;
    @Builder.Default
    private String tokenType = "Bearer";
    private Long expiresInMs;
    private Long userId;
    /** ID del registro Personal asociado al usuario (usado para filtrar citas, consultas, etc.) */
    private Long personalId;
    private String email;
    private String nombreCompleto;
    /** Rol primario (SUPERADMINISTRADOR > ADMIN > primero) — retrocompatibilidad. */
    private String role;
    /** Todos los roles asignados al usuario. */
    private List<String> roles;
    /**
     * Rol activo seleccionado por el usuario en la sesión actual.
     * Inicialmente igual a {@code role}; puede cambiarse en el frontend sin necesidad
     * de reautenticación (se persiste en localStorage).
     */
    private String rolActivo;
    /** Esquema del tenant (empresa) para enviar en header X-Tenant-Schema en peticiones */
    private String schema;
    /** Nombre de la empresa (razón social) del tenant actual, para mostrar en la UI */
    private String empresaNombre;

    /**
     * UUID del logo de la empresa (si existe en archivo_almacenamiento).
     * El frontend construye la URL directa: {@code /api/archivos/{empresaLogoUuid}}.
     * Al incluirlo en el login, el logo está disponible inmediatamente sin un GET extra.
     */
    private String empresaLogoUuid;
}

File: sesa-mobile/lib/core/models/login_response.dart
// Autor: Ing. J Sebastian Vargas S
// Modelo de respuesta del endpoint POST /api/auth/login

class LoginResponse {
  final String accessToken;
  final String tokenType;
  final int expiresInMs;
  final int userId;
  final String email;
  final String nombreCompleto;
  final String role;
  final String schema;
  final String empresaNombre;

  const LoginResponse({
    required this.accessToken,
    required this.tokenType,
    required this.expiresInMs,
    required this.userId,
    required this.email,
    required this.nombreCompleto,
    required this.role,
    required this.schema,
    required this.empresaNombre,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) => LoginResponse(
    accessToken   : json['accessToken']    as String,
    tokenType     : json['tokenType']      as String? ?? 'Bearer',
    expiresInMs   : json['expiresInMs']    as int?    ?? 86400000,
    userId        : (json['userId'] as num).toInt(),
    email         : json['email']          as String,
    nombreCompleto: json['nombreCompleto'] as String,
    role          : json['role']           as String,
    schema        : json['schema']         as String,
    empresaNombre : json['empresaNombre']  as String? ?? 'SESA Global',
  );

  Map<String, dynamic> toJson() => {
    'accessToken'  : accessToken,
    'tokenType'    : tokenType,
    'expiresInMs'  : expiresInMs,
    'userId'       : userId,
    'email'        : email,
    'nombreCompleto': nombreCompleto,
    'role'         : role,
    'schema'       : schema,
    'empresaNombre': empresaNombre,
  };
}
